systemtests/tests/ktls-test/CMakeLists.txt

```diff
@@ -0,0 +1,57 @@
+#   BAREOS® - Backup Archiving REcovery Open Sourced
+#
+#   Copyright (C) 2023-2023 Bareos GmbH & Co. KG
+#
+#   This program is Free Software; you can redistribute it and/or
+#   modify it under the terms of version three of the GNU Affero General Public
+#   License as published by the Free Software Foundation and included
+#   in the file LICENSE.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#   Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#   02110-1301, USA.
+
+get_filename_component(BASENAME ${CMAKE_CURRENT_BINARY_DIR} NAME)
+
```

Comment:
Great that we have the ktls tests now. As I understood, it should also work on FreeBSD

Reply:
On my laptop, the KTLS tests all failed unless I did `modprobe tls` manually. Maybe we can detect that somehow?

Reply:
Yes modprobe is required.  On FreeBSD: Im not sure if it works with what we have.  There is probably a socket option we have to set.   I can look  into that if we also want to have it on free bsd.
Not sure if we can detect kernel modules.   Ill definitely add it to the documentation.